src/Symfony/Bundle/FrameworkBundle/DependencyInjection/Compiler/CachePoolClearerPass.php

```diff
@@ -0,0 +1,40 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Bundle\FrameworkBundle\DependencyInjection\Compiler;
+
+use Symfony\Component\DependencyInjection\Compiler\CompilerPassInterface;
+use Symfony\Component\DependencyInjection\ContainerBuilder;
+use Symfony\Component\DependencyInjection\Reference;
+
+/**
+ * @author Nicolas Grekas <p@tchwork.com>
+ */
+final class CachePoolClearerPass implements CompilerPassInterface
+{
+    /**
+     * {@inheritdoc}
+     */
+    public function process(ContainerBuilder $container)
+    {
+        foreach ($container->findTaggedServiceIds('cache.pool') as $id => $attributes) {
+            foreach (array_reverse($attributes) as $attr) {
+                if (isset($attr['clearer'])) {
+                    $clearer = $container->getDefinition($attr['clearer']);
+                    $clearer->addMethodCall('addPool', array(new Reference($id)));
+                }
+                if (array_key_exists('clearer', $attr)) {
```

Comment:
I don't understand logic behind this code (`array_reverse` + `break`). Do you want to emulate overriding of the tag? Does it mean that a service cannot have 2 different clearers?